paddle/fluid/operators/mkldnn/concat_mkldnn_op.cc

```diff
@@ -120,6 +158,16 @@ template <typename T>
 class ConcatMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
  public:
   void Compute(const paddle::framework::ExecutionContext& ctx) const override {
+    bool is_INT8 =
+        std::is_same<T, int8_t>::value || std::is_same<T, uint8_t>::value;
```

Comment:
Wouldn't it be better to use template specialisation for `int8_t` and `uint8_t` for Compute method? That way, the method to call would be known at the compile time and we'd avoid both RTTI and a branch which are a bit performance costy.

Reply:
@Sand3r- No, compile time does not know the data type. You may refer to INT8 kernel at conv_mkldnn_op.cc as well.

Reply:
It actually does, because it happens during the registration of an op. Then during the runtime an appropriate instance of such template method would be called.

Reply:
Thanks @Sand3r- . How about adding a JIRA internally to experiment your suggestion, with other INT8 kernels, like Conv? With that, we can have a unified solution.